File: src/windows/magma/api/collect/list/ListCollector.c
#include "./ListCollector.h"
export class ListCollector<T> implements Collector<T, List<T>> {
}

List<T> createInitial() {
	return Lists/*auto*/.empty(/*auto*/);
}
List<T> fold(List<T> current, T element) {
	return current/*List<T>*/.addLast(element/*T*/);
}

File: packages/compiler-java/src/main/java/magmac/api/iter/collect/ListCollector.java
package magmac.api.iter.collect;

import magmac.api.collect.list.List;
import magmac.api.collect.list.Lists;

/**
 * Collector that accumulates elements into a {@link magmac.api.collect.list.List}.
 */

public record ListCollector<T>() implements Collector<T, List<T>> {
    @Override
    public List<T> createInitial() {
        return Lists.empty();
    }

    @Override
    public List<T> fold(List<T> current, T element) {
        return current.addLast(element);
    }
}
